Source/SwiftLintFramework/Rules/Lint/DuplicateConditionsRule.swift

```diff
@@ -0,0 +1,214 @@
+import SwiftSyntax
+
+struct DuplicateConditionsRule: SwiftSyntaxRule, ConfigurationProviderRule {
+    var configuration = SeverityConfiguration(.warning)
+
+    init() {}
+
+    static let description = RuleDescription(
+        identifier: "duplicate_conditions",
+        name: "Duplicate Conditions",
+        description: "Duplicate sets of conditions in the same branching statement should be avoided",
```

Comment:
```suggestion
        description: "Duplicate sets of conditions in the same branch condition should be avoided",
```

Reply:
I don't think my original choice of phrasing was very good, but I'm also not certain about `branch condition`; to me this means, for example, `x < 5` in `if x < 5 { foo() } else { bar() }`. What do you think about `branch instruction` instead?

Reply:
Perfect :)